src/dynamics/video_player/player/player.js

```diff
@@ -585,8 +586,16 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                             if (thisPlaylist.length === 1 && thisPlaylist[0].title === this.get('title')) return;
                             const showNextTime = Number(this.get("shownext")) || 0;
                             const engageTime = showNextTime + (Number(this.get("noengagenext")) || 0);
-                            if (position > showNextTime && showNextTime && !this.get("next_active")) {
-                                this.set("next_active", true);
+                            const duration = this.get("duration");
+                            const remainingTime = (duration - position);
+                            const minIntervalNextWidget = this.get("nextwidgetminintervalbeforeend");
+                            const shallActivateNext = minIntervalNextWidget < 0 ? true :
+                                (duration - showNextTime) >= (Number(this.get("noengagenext")) || 0);
+
+                            if (showNextTime && position > showNextTime && !this.get("next_active")) {
+                                if (minIntervalNextWidget > 0 && remainingTime >= minIntervalNextWidget || shallActivateNext) {
+                                    this.set("next_active", true);
+                                }
```

Comment:
Hi, @Josewd! I think this makes it easier to understand the logic, let me know what you think.
```suggestion
                            let minDuration = this.get("mindurationnext");
                            if (!minDuration) minDuration = engageTime;
                            if (this.get("duration") >= minDuration && showNextTime && position > showNextTime && !this.get("next_active")) {
                                this.set("next_active", true);
```